trunk/SUAVE/Methods/Propulsion/electric_motor_sizing.py

```diff
@@ -80,8 +88,49 @@ def size_from_mass(motor,mass):
     B_i0 = 0.2   * Units['amp*(ohm**0.6)']
     
     # Do the calculations from the regressions
-    kv = B_KV/mass
+    kv   = B_KV/mass
     res  = B_RA/(kv**2.)
-    i0   =  B_i0/(res**0.6)
+    i0   = B_i0/(res**0.6) 
+
+    # Unpack the motor
+    motor.resistance      = res 
+    motor.no_load_current = i0  
+    motor.speed_constant  = kv    
 
     return motor
+
+
+def compute_optimal_motor_parameters(motor,prop):
+    ''' Optimizes the motor to obtain the best combination of speed constant and resistance values
+    by essentially you are sizing the motor for a design RPM value. Note that this design RPM 
+    value can be compute from design tip mach  
+    
+    Source:
+     
+    
+    Inputs:
+    motor    (to be modified)
+    
+    Outputs:
+    motor.
+      speed_constant     [untiless]
+      no_load_current    [amps]
+    '''    
+    
+    io                   = motor.no_load_current
+    v                    = motor.nominal_voltage 
+    omeg                 = prop.angular_velocity
+    etam                 = motor.efficiency
+    start_kv             = 1
```

Comment:
don't hard code these. 15 kv is actually quite low for some applications.

Reply:
See comment below